test/unit/general/iotam-autoregistration-test.js

```diff
@@ -366,10 +458,12 @@ describe('NGSI-v1 - IoT Manager autoregistration', function() {
 
         it('should update the registration in the IoT Manager', function(done) {
             request(optionsCreationStatic, function(error, result, body) {
+              request(optionsCreationStaticconfigGroups, function(error, result, body) {
                 should.not.exist(error);
                 iotamMock.done();
                 done();
             });
         });
     });
```

Comment:
Instead of "mixing" old tests with new tests, I'd suggest to do an independent test. It is clearer and will allow in the future easily remove deprecated stuff if we want.

I mean:

1. Remove this line, leaving the old test as it was before
2. Create a new parallel test case for the new functionality, for instance (note the "using configuration group" in the description):

```
        it('should update the registration in the IoT Manager using configuration group', function(done) {
            request(optionsCreationStaticconfigGroups, function(error, result, body) {              
                should.not.exist(error);
                iotamMock.done();
                done();
            });
        });
```

3. Put the `// This test will be removed if at the end the /iot/services API (now deprecated) is removed` comment just above the `it('should update the registration in the IoT Manager', function(done) {` of the old test, taking account indentation (ie. `//` should be aligned with `it`).

Please do the same for every new test in every file.

Reply:
@fgalan As per your suggestion, if I add a new test (https://github.com/telefonicaid/iotagent-node-lib/pull/780/commits/f4591ccd9916179fa46d0bf3021a9d20d8abaa79), travis is giving errors. Would it not be feasible to add the new functionality in old tests itself and whenever there might be need to remove the old functionality we could just remove that section from the tests. What are your thoughts ?

Reply:
@fgalan Any updates ?

Reply:
Looking to the current version of the PR, I see you have opted to put new tests on new files (e.g. test/unit/general/iotam-autoregistration-configurationGrouptest.js). That's nice, because it provides a good separation (and, in the future it is just a matter or remove the test/unit/general/iotam-autoregistration-test.js file corresponding to the old endpoints, when time comes).

Failing tests are not necessarily a bad thing. They could be pointing out some flaw in the implementation. And it is always better to have tests separated, so they can focus in the functionality are supposed to tests. I mean, if you have 2 separate functionalities to tests it is better to have 2 test cases, each one focusing in one functionality than just 1 test case trying to cover both them.

Thus, my adive is you to review the new tests and find out why they are failing. It could be a problem in the test specification or in the functionality and the cause needs to be researched.

Reply:
I have added the new tests and the implementation functionality that was missing (https://github.com/telefonicaid/iotagent-node-lib/pull/780/commits/877c8b02e5353a104df5f245144042c5eebfb1dd). However travis has stopped running. Could you re-run the travis to check where the tests have passed,